### PR TITLE
[Synapse] Add support for Log Analytics and Event Hub in workspace auditing and sql pool auditing

### PR DESCRIPTION
**Pull request by wonner**
## Description

Added some parameters in cmdlet `Set-AzSynapseSqlAuditSetting` and `Set-AzSynapseSqlPoolAuditSetting` to support Log Analytics and Event Hub in workspace and sql pool level auditing.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/989
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by idear1203 — APPROVED** [review]

File: src/Synapse/Synapse.Test/ScenarioTests/SynapseTestBase.cs [hunk 1]
```diff
@@ -51,6 +55,14 @@ public class SynapseTestBase : RMTestBase
 
         public StorageManagementClient StorageManagementClient { get; private set; }
 
+        public CommonMonitor.MonitorManagementClient CommonMonitorManagementClient { get; private set; }
```
Comment (VeryEarly) [inline src/Synapse/Synapse.Test/ScenarioTests/SynapseTestBase.cs:58]: please add reference to corresponding psd1 file in line 119

File: src/Synapse/Synapse/ChangeLog.md [hunk 2]
```diff
@@ -19,6 +19,8 @@
 -->
 ## Upcoming Release
 * Added parameter `-ManagedResourceGroupName` for the `New-AzSynapseWorkspace` cmdlet
+* `Set-AzSynapseSqlAuditSetting` and `Set-AzSynapseSqlPoolAuditSetting` now supports event hub and log analytics
```
Comment (VeryEarly) [inline src/Synapse/Synapse/ChangeLog.md:22]:
```suggestion
* Added support for event hub and log analytics to `Set-AzSynapseSqlAuditSetting` and `Set-AzSynapseSqlPoolAuditSetting`
```

**Review by VeryEarly — CHANGES_REQUESTED** [review]

**Comment by VeryEarly** [comment]
please fix and re-record test cases

**Comment by wonner** [comment]
@VeryEarly I have already updated the code according to your suggestions and fixed the test cases, please kindly review again. Thanks!

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**